schemas/input/csl-data.json

```
@@ -468,8 +473,19 @@
         }
       ]
     },
+    "edtf-datatype": {
+      "title": "EDTF datatype pattern",
+      "description": "CSL input supports EDTF, validated against this regular expression.",
+      "type": "string",
+      "pattern": "[0-9,-,%,~,X,?,..,\/]"
```

this only tests valid characters; would like tighter ideally

---

This regex doesn't look correct to me. `[0-9,-,%,~,X,?,..,\/]` only matches a single character (anything in the brackets).

You probably want something more like https://regex101.com/r/IZJFVM/1

If you describe what you want the regex to test I can help your with the syntax.

---

Thanks @dhimmel.

Your regex is good, but the final version will depend on answers to the compliance questions. For example, it doesn't currently support sets, but I'm not convinced it should.

I've updated the main post with examples, which I've used to test your mod. Here's [your's with the added test data](https://regex101.com/r/IZJFVM/2).

---

```suggestion
      "pattern": "^[0-9\-~X?.\\\/]+$"
```

So, depending on any final tweaks, this?

---

> If you describe what you want the regex to test I can help your with the syntax.

@dhimmel - I think we want to exclude the set feature; here's an example: `{2010-03,2010-05,2010-03..2010-05}`.

So I guess that would mean not accepting commas or brackets. I'm unclear on why the above does accept them (at least in my testing).

---

Actually, I think the validator I'm testing with isn't using the regex. So I think what I have now should work.